test/test_facevalues.jl

```diff
@@ -95,7 +96,7 @@ for (scalar_interpol, quad_rule) in (
         # test copy
         fvc = copy(fv)
         @test typeof(fv) == typeof(fvc)
-        for fname in fieldnames(typeof(fv))
+        Ferrite.getrefshape(func_interpol) ∈ (RefPrism, RefPyramid)|| for fname in fieldnames(typeof(fv))
```

Comment:
What is the problem here?

Reply:
It's that the copy test fails for matrices with NaN entries, and for cells with faces that have different number of dofs/quadpoints we get NaN entries for the extra entries for the faces with ndofs/nqp < max ndofs/nqp

Reply:
We should be able to fix this by introducing some "matrix"-like struct with varying row/col size. I.e. something like
```julia
struct VaryingMatrix{TOV, TDV}
   offsets::TIV
   data::TDV
end
```
where the access with two indices first looks up the offset in the offset vector and the second index computes the actual element in the data vector via data[offset_index[i]+j], where the offsets start at 0. We should then start using this all over the codebase. Does this make sense? cc @fredrikekre separate PR maybe?

Reply:
I was thinking of using a vector of matrices instead of a 3-D array `Array{T, 3}` but I'm not sure what the performance impact would be due to changing memory layout

Reply:
Yea, we should try to keep the memory continuous here, because there is potentially quite a big performance hit, depending on the system.